server/events/command_runner.go

```diff
@@ -94,6 +94,18 @@ type DefaultCommandRunner struct {
 	LegacyLogger                  logging.SimpleLogging
 }
 
+func (p *DefaultCommandRunner) updateCombinedCommitStatusAndLogError(ctx *command.Context, status models.CommitStatus, cmdName fmt.Stringer) {
+	baseRepo := ctx.Pull.BaseRepo
+	pull := ctx.Pull
+	statusID, err := p.CommitStatusUpdater.UpdateCombined(context.TODO(), baseRepo, pull, status, cmdName, ctx.StatusID)
```

Comment:
Why are you using `context.TODO()`?  This is available in the calling function, just pass that through.